### PR TITLE
Relax ghproxy alert rules by extending duration of status code starting with 4 or 5

### PR DESCRIPTION
**Pull request by chaodaiG**
Part of: #21138 

Currently prow-alerts channel is pretty much filled with warnings from ghproxy, they might provide some value but it's so spammy that if any other useful alerts were surfaced they would easily be overlooked.

Simply inspecting the warnings, majority of them were 422 and 405 codes. And some of them were actually expected, for example:

Expecting 422 from https://github.com/kubernetes/test-infra/blob/7f3fe6c5ffbd8a665d68e38b6ddbd865e06716a6/prow/github/client.go#L2707
Expecting 405 from https://github.com/kubernetes/test-infra/blob/7f3fe6c5ffbd8a665d68e38b6ddbd865e06716a6/prow/github/client.go#L3472

Extending to 30 minutes and hopefully it would dilute the impact enough to eliminate the noise. If this doesn't work will need to figure out how to correctly set warnings for these two status codes

### CODE REVIEW

File: config/prow/cluster/monitoring/mixins/prometheus/ghproxy_alerts.libsonnet [hunk 1]
```diff
@@ -9,7 +9,7 @@
           {
             alert: 'ghproxy-specific-status-code-abnormal',
             expr: |||
-              sum(rate(github_request_duration_count{status=~"[45]..",status!="404",status!="410"}[5m])) by (status,path) / ignoring(status) group_left sum(rate(github_request_duration_count[5m])) by (path) * 100 > 10
+              sum(rate(github_request_duration_count{status=~"[45]..",status!="404",status!="410"}[30m])) by (status,path) / ignoring(status) group_left sum(rate(github_request_duration_count[5m])) by (path) * 100 > 10
```
Comment (cjwagner) [inline config/prow/cluster/monitoring/mixins/prometheus/ghproxy_alerts.libsonnet:12]: I skimmed through some of the #prow-alerts channel and didn't see any 5xx alerts.  WDYT about keeping the tighter alerting window for 5xx and only loosening 4xx?
5mins to 30mins is a pretty big change and delays the alerts for so long that they we'd be likely to notice serious problems ourselves before the alert fires. Could we choose a smaller time window if we also increased the error threshold to more than 10%?

Comment (chaodaiG) [reply config/prow/cluster/monitoring/mixins/prometheus/ghproxy_alerts.libsonnet]: good point about 5xx, updated to keep 5xx still alert with 5m window events.

I have seen 100% for example https://kubernetes.slack.com/archives/CQP3RC68Y/p1614920511052900, which imo would be pretty hard to brought down to 20% even with doubled time, so I think diluting it out is more effective, still finger crossed on this though

File: config/prow/cluster/monitoring/mixins/prometheus/ghproxy_alerts.libsonnet [hunk 2]
```diff
@@ -19,9 +19,33 @@
...
+              severity: 'warning',
+            },
+            annotations: {
+              message: '{{ $value | humanize }}%% of all API requests through the GitHub proxy are errorring with code {{ $labels.status }}. Check %s.' % [monitoringLink('/d/%s/github-cache?orgId=1&refresh=1m&fullscreen&panelId=8' % [dashboardID], 'the ghproxy dashboard')],
+            },
+          },
+          {
+            alert: 'ghproxy-specific-status-code-4xx',
+            expr: |||
+              sum(rate(github_request_duration_count{status=~"4..",status!="404",status!="410"}[30m])) by (status,path) / ignoring(status) group_left sum(rate(github_request_duration_count[5m])) by (path) * 100 > 10
```
Comment (cjwagner) [inline config/prow/cluster/monitoring/mixins/prometheus/ghproxy_alerts.libsonnet:36]: The second time interval needs to be updated to `30m` as well. Same with the other 4xx alert.

Comment (chaodaiG) [reply config/prow/cluster/monitoring/mixins/prometheus/ghproxy_alerts.libsonnet]: good catch, done

**Review by cjwagner — APPROVED** [review]

**Pull request merged**